### PR TITLE
PR to populate disable detectors list

### PR DESCRIPTION
**Pull request by Omotola**
Readded changes to populate disabled detectors list and added some unit tests. Also modified `ScanResult` contract to help with the unit tests.

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs [hunk 1]
```diff
@@ -38,14 +38,17 @@ public async Task<ScanResult> ExecuteScanAsync(ScanSettings settings)
         using var scope = this.logger.BeginScope("Executing BCDE scan");
 
         var detectorRestrictions = this.GetDetectorRestrictions(settings);
-        var detectors = this.detectorRestrictionService.ApplyRestrictions(detectorRestrictions, this.detectors).ToImmutableList();
+        var detectors = this.detectors.ToList();
```
Comment (Omotola) [inline src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs:41]: Added this line to convert to a list cause the tests were failing with IEnumerable

File: src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs [hunk 2]
```diff
@@ -38,14 +38,17 @@ public async Task<ScanResult> ExecuteScanAsync(ScanSettings settings)
...
+        var restrictedDetectors = this.detectorRestrictionService.ApplyRestrictions(detectorRestrictions, detectors).ToImmutableList();
+        detectorRestrictions.DisabledDetectors = detectors.Except(restrictedDetectors).ToList();
 
         this.logger.LogDebug("Finished applying restrictions to detectors.");
 
-        var processingResult = await this.detectorProcessingService.ProcessDetectorsAsync(settings, detectors, detectorRestrictions);
+        var processingResult = await this.detectorProcessingService.ProcessDetectorsAsync(settings, restrictedDetectors, detectorRestrictions);
```
Comment (grvillic) [inline src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs:47]: nit: This naming seems a bit confusing. We need to process all the detectors. This is just a subset of detectors, but naming it like this + the `detectorRestrictions` next to it makes it confusing as to what are we actually passing.

File: src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs (hunk 1 above)
Comment (grvillic) [reply src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs:41]: nit: but I think the tests were not mocked properly, we should have to enlist the entries to make then work. Just your check may need to be looking for equivalent objects not the same exact object.

**Review by grvillic — APPROVED** [review]

File: src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs (hunk 1 above)
Comment (Omotola) [reply src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs:41]: The initial one that fails is actually only checking if `DetectorsNotInRun` is not empty image

**Pull request merged**